File: services/rendering/src/smoke.ts
/**
 * Rendering service smoke test.
 * Invoked by: `pnpm --filter @inzohra/rendering smoke`
 */
function main(): void {
  const required = ["DATABASE_URL", "S3_ENDPOINT"];
  const missing = required.filter((k) => !process.env[k]);
  if (missing.length > 0) {
    console.error(`RENDERING SMOKE: FAIL — missing env: ${missing.join(", ")}`);
    process.exit(1);
  }

  // Verify dependencies import.
  // eslint-disable-next-line @typescript-eslint/no-require-imports
  require("pdfkit");
  // eslint-disable-next-line @typescript-eslint/no-require-imports
  require("docx");

  console.log("RENDERING SMOKE: OK");
}

main();

File: services/measurement/app/smoke.py
"""Measurement service smoke test."""
from __future__ import annotations
import os
import sys


def main() -> None:
    required = ["DATABASE_URL", "REDIS_URL"]
    missing = [k for k in required if not os.getenv(k)]
    if missing:
        print(f"MEASUREMENT SMOKE: FAIL — missing env: {missing}")
        sys.exit(1)

    import cv2  # noqa: F401
    import numpy as np  # noqa: F401
    import shapely  # noqa: F401
    import fitz  # noqa: F401

    print("MEASUREMENT SMOKE: OK")


if __name__ == "__main__":
    main()
